feat(formatter): Export doc() API to inspect IR in example

leaysgur opened the pull request:
```sh
❯ cargo run -p oxc_formatter --example formatter t.ts --ir
    Finished `dev` profile [unoptimized] target(s) in 0.05s
     Running `target/debug/examples/formatter t.ts --ir`
[
  StaticText("import"),
  Space,
  StaticText("{"),
  Space,
  DynamicText("useState"),
  Space,
  StaticText("}"),
  Space,
  StaticText("from"),
  Space,
  LocatedTokenText("\"react\""),
  StaticText(";"),
  Line(Empty),
  StaticText("export"),
  Space,
  StaticText("function"),
  Space,
  DynamicText("useData"),
  Tag(StartGroup(Group { id: None, mode: Cell { value: Flat } })),
  Tag(EndGroup),
  Tag(StartGroup(Group { id: None, mode: Cell { value: Flat } })),
  StaticText("("),
  Tag(StartGroup(Group { id: None, mode: Cell { value: Flat } })),
  Tag(EndGroup),
  StaticText(")"),
  Tag(EndGroup),
  Space,
  Space,
  StaticText("{"),
  Tag(StartIndent),
  Line(Hard),

...

  StaticText("}"),
  Line(Hard),
]
```

leaysgur commented:
* **#14068** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14068?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14068?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=14068).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing reviewed (approved):
It looks good for now. I will refactor it when the TypeScript support is complete.

Dunqing commented:
### Merge activity

* **Sep 24, 6:22 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
feat(formatter): Export doc() API to inspect IR in example (#14068)

```sh
❯ cargo run -p oxc_formatter --example formatter t.ts --ir
    Finished `dev` profile [unoptimized] target(s) in 0.05s
     Running `target/debug/examples/formatter t.ts --ir`
[
  StaticText("import"),
  Space,
  StaticText("{"),
  Space,
  DynamicText("useState"),
  Space,
  StaticText("}"),
  Space,
  StaticText("from"),
  Space,
  LocatedTokenText("\"react\""),
  StaticText(";"),
  Line(Empty),
  StaticText("export"),
  Space,
  StaticText("function"),
  Space,
  DynamicText("useData"),
  Tag(StartGroup(Group { id: None, mode: Cell { value: Flat } })),
  Tag(EndGroup),
  Tag(StartGroup(Group { id: None, mode: Cell { value: Flat } })),
  StaticText("("),
  Tag(StartGroup(Group { id: None, mode: Cell { value: Flat } })),
  Tag(EndGroup),
  StaticText(")"),
  Tag(EndGroup),
  Space,
  Space,
  StaticText("{"),
  Tag(StartIndent),
  Line(Hard),

...

  StaticText("}"),
  Line(Hard),
]
```
